cmake/Gems.cmake

```diff
@@ -76,6 +92,13 @@ macro(o3de_gem_setup default_gem_name)
     o3de_read_json_key(gem_version ${gem_json} "version")
 
     o3de_restricted_path(${gem_json} gem_restricted_path gem_parent_relative_path)
+    if(gem_restricted_path)
+        message(STATUS "Restricted found in ${gem_json}, using: ${gem_restricted_path}")
```

Comment:
Do we want these to be STATUS?  They will show up unconditionally during every configure.